backend/src/apiserver/client_manager/client_manager.go

```diff
@@ -178,12 +208,62 @@ func (c *ClientManager) init() {
 	// UUID generator
 	c.uuid = util.NewUUIDGenerator()
 
+	var controllerClient ctrlclient.Client
+
+	var pipelineStoreForRef storage.PipelineStoreInterface
+
+	if options.UsePipelineKubernetesStorage {
```

Comment:
Just a note that this new flag should likely be used to determine if the validating webhook in #11684 should be enabled.